### PR TITLE
Seal up protected

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring
Breaking change: yes

Depends on:
- https://github.com/phpmd/phpmd/pull/1144

Seal up properties and functions that do not need to be protected. This will allow us more flexibility in terms of refactoring in the future. We can always expose things that makes sens rather then everything by default.
This also lets tools better detected unreachable conditions or dead code.

### CODE REVIEW

File: src/main/php/PHPMD/RuleSetFactory.php [hunk 1]
```diff
@@ -168,16 +168,19 @@ private function createRuleSetFileName($ruleSetOrFileName)
...
     private static function listRuleSetsInDirectory($directory)
     {
         $ruleSets = [];
         if (is_dir($directory)) {
-            foreach (scandir($directory) as $file) {
-                $matches = [];
-                if (is_file($directory . $file) && preg_match('/^(.*)\.xml$/', $file, $matches)) {
-                    $ruleSets[] = $matches[1];
+            $filesPaths = scandir($directory);
+            if ($filesPaths !== false) {
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/RuleSetFactory.php:178]: Maybe we could use `?: []` instead of the `if`

**Review by kylekatarnls — APPROVED** [review]

File: src/main/php/PHPMD/Cache/ResultCacheFileFilter.php [hunk 2]
```diff
@@ -65,13 +65,15 @@ public function accept($relative, $absolute): bool
         }
 
         // Determine if file was modified since last analyse
-        if ($this->state === null) {
+        if ($this->state === null || $hash === false) {
             $isModified = true;
         } else {
             $isModified = $this->state->isFileModified($filePath, $hash);
         }
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply src/main/php/PHPMD/Cache/ResultCacheFileFilter.php:72]: Sure, will do a follow up.

[the remark this answers is not in the document]
Comment (AJenbo) [reply src/main/php/PHPMD/Cache/ResultCacheFileFilter.php:72]: Sure, will do a follow up.

Here you got: https://github.com/phpmd/phpmd/pull/1148

File: src/main/php/PHPMD/RuleSetFactory.php (hunk 1 above)
Comment (AJenbo) [reply src/main/php/PHPMD/RuleSetFactory.php:178]: Done: https://github.com/phpmd/phpmd/pull/1148

**Pull request merged**